internal/reporter.go

```diff
@@ -90,15 +94,26 @@ func (reporter reporter) buildRequest(format string, body []byte) (*http.Request
 	return req, nil
 }
 
-func (reporter reporter) ReportEvent(event string) (*http.Response, error) {
+func (reporter reporter) reportEvent(event string) (*http.Response, error) {
 	if event == "" {
 		return nil, formatError
 	}
 
 	apiURL := reporter.serverURL + eventEndpoint
-	req, err := http.NewRequest("POST", apiURL, strings.NewReader(event))
+	var body io.Reader
```

Comment:
I think that the way the code is prior to this PR reflects some kind of logical error.

As written, it will add the `Content-Encoding: gzip` header to event requests that aren't gzipped.

I think that the code is trying to make sure that all requests directly posting to a Wavefront host use `gzip`, but forgot to do the encoding step.

Events are odd - the `Direct` code path ingests them as JSON, but the proxy code path expects lines starting with `@Event`.

I think that we might as well `gzip` all of these all the time.